functions/slack/main.py

```
@@ -0,0 +1,100 @@
+
+
+# [START functions_verify_webhook]
+def verify_web_hook(form):
+    if not form or form.get('token') != config['SLACK_TOKEN']:
+        raise ValueError('Invalid request/credentials.')
+# [END functions_verify_webhook]
+
+
+# [START functions_slack_format]
+def format_slack_message(query, response):
+    entity = None
+    if response and response.get('itemListElement') is not None and \
+       len(response['itemListElement']) > 0:
+        entity = response['itemListElement'][0]['result']
+
+    message = {
+        'response_type': 'in_channel',
+        'text': f'Query: {query}',
+        'attachments': []
+    }
+
+    attachment = {}
+    if entity:
+        name = entity.get('name', '')
+        description = entity.get('description', '')
+        detailed_desc = entity.get('detailedDescription', {})
+        url = detailed_desc.get('url')
+        article = detailed_desc.get('articleBody')
+        image_url = entity.get('image', {}).get('contentUrl')
+
+        attachment['color'] = '#3367d6'
+        if name and description:
+            attachment['title'] = f'{entity["name"]}: {entity["description"]}'
+        elif name:
+            attachment['title'] = name
+        if url:
+            attachment['title_link'] = url
+        if article:
+            attachment['text'] = article
+        if image_url:
+            attachment['image_url'] = image_url
+    else:
+        attachment['text']: 'No results match your query.'
+    message['attachments'].append(attachment)
+
+    return message
+# [END functions_slack_format]
+
+
+# [START functions_slack_request]
+def make_search_request(query):
+    try:
+        req = kgsearch.entities().search(query=query, limit=1)
+        res = req.execute()
+        print(res)
+        return format_slack_message(query, res)
+    except apiclient.errors.Error as err:
+        raise ValueError(str(err))
+# [END functions_slack_request]
+
+
+# [START functions_slack_search]
+def kg_search(request):
```

Where is this function used, is it an HTTP handler for a Function? In that case Flask has a good way of returning a 405 error, and we don't have to catch errors and print them -- just let them bubble up.